### ISSUE TITLE
Using Observatory behind corporate proxy ?

### ISSUE THREAD

**Issue by unknownboy** [mozilla/http-observatory#89]
Hi,

First, thank you for this great project !

I want to use http-observatory for internal corporate tests. So, I just installed the tool but when starting it, the code from utils.py try to retrieving the Google HSTS Preload list from "https://chromium.googlesource.com/chromium/src/net/+/master/http/transport_security_state_static.json?format=TEXT". As we are behind a proxy for Internet access, i was trying to set up ENV variables but it's not working. I used http_proxy and https_proxy pointing to a local cntml proxy. 
I also tried to modify utils.py in order to use a proxy configuration, without succes (python noob writing).

``` python
proxiesList = {
  'https': 'http://localhost:3128'
}

# Download the Google HSTS Preload List
try:
    print('Retrieving the Google HSTS Preload list', file=sys.stderr)
    r = b64decode(requests.get(HSTS_URL,proxies=proxiesList,timeout=5).text).decode('utf-8').split('\n')
```

What am I missing ?

**Comment by marumari** [comment]
Does a proxy on your localhost have access to reach out across the intranet?

Instead of modifying the code, you're probably better off setting HTTP_PROXY and HTTPS_PROXY as environmental variables.

**Comment by unknownboy** [comment]
The proxy on localhost has access to internet.
HTTP_PROXY and HTTPS_PROXY are already set.
Other tools that are using the ENV VAR are working well and can access internet.

**Comment by floatingatoll** [comment]
From the command line where the two environment variables are already set,
does this command return a valid result?

$ curl -vsI
https://chromium.googlesource.com/chromium/src/net/+/master/http/transport_security_state_static.json
>/dev/null

When you say "not working", what exception is shown by the try/catch block
when the requests.get fails?

**Comment by unknownboy** [comment]
curl is working (200 return code).
When i say "not working", the code is stuck on "Retrieving the Google HSTS Preload list". It seems like there is no timeout (that's why i was trying to add "timeout=5" to "requests.get()").

**Comment by marumari** [comment]
With those environmental variables set, what happens if you use requests directly from the Python CLI?

```python
>>> import requests
>>> requests.get('https://chromium.googlesource.com/chromium/src/net/+/master/http/transport_security_state_static.json')
```

**Comment by unknownboy** [comment]
It's working from Python CLI :
>>> requests.get('https://chromium.googlesource.com/chromium/src/net/+/master/http/transport_security_state_static.json')
**<Response [200]>**

**Comment by floatingatoll** [comment]
#160 was committed recently, which makes the preload list local rather than remote. I was having the same issue fetching it *without* a corporate proxy in the way, but now that it's local, the problem is sidestepped entirely. I'm going to mark this as closed by #160, but please feel free to indicate otherwise if it's not resolved for you after updating to the latest code.
